src/decision/location_selection.jl

```diff
@@ -0,0 +1,356 @@
+    rank_frequencies = NamedDimsArray(zeros(length(ranks.timesteps), length(ranks.sites), length(ranks.sites)), timesteps=ranks.timesteps, sites=ranks.sites, ranks=1:length(ranks.sites))
+    return ranks_to_frequencies(ranks, rank_frequencies; n_ranks=n_ranks)
+end
+function ranks_to_frequencies_ts(
+    ranks::NamedDimsArray, iv_type::String; n_ranks=length(ranks.sites)
+)
+    selected_ranks = _get_iv_type(ranks, iv_type)
+    return ranks_to_frequencies_ts(selected_ranks; n_ranks=n_ranks)
+end
+function ranks_to_frequencies_ts(
+    rs::ResultSet, iv_type::String; n_ranks=length(rs.ranks.sites)
+)
+    selected_ranks = _get_iv_type(rs.ranks, iv_type)
+    return ranks_to_frequencies_ts(selected_ranks; n_ranks=n_ranks)
+end
+
+"""
+    location_selection_frequencies(ranks::NamedDimsArray, iv_type::String; n_loc_int=5, ind_metrics=collect(1:length(ranks.scenarios)))
+    location_selection_frequencies(rs::ResultSet, iv_type::String; n_loc_int=5, ind_metrics=collect(1:length(ranks.scenarios)))
+ 
+Post-processing function for intervention logs. Calculates the frequencies with which locations were selected for a particular intervention,
+for a selection of scenarios (e.g. selected robust scenarios).
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `rs` : ADRIA result set.
+- `ind_metrics` : Indices for selected scenarios (such as robust scenarios).
+- `iv_type` : indicates intervention log to use ("seed", "shade" or "fog").
+- `n_loc_int` : number of locations which are intervened at for each intervention decision.
+
+# Returns 
+- Counts for location selection at each location in the domain.
+"""
+function location_selection_frequencies(
+    ranks::NamedDimsArray,
+    iv_type::String;
+    n_loc_int::Int64=5,
+    ind_metrics::Vector{Int64}=_get_scen_ids(ranks),
+)
+    ranks_frequencies = ranks_to_frequencies(ranks[scenarios=ind_metrics], iv_type; n_ranks=n_loc_int)
+    loc_count = sum(ranks_frequencies[ranks=1:n_loc_int], dims=2)[ranks=1]
+
+    return loc_count
+end
+function location_selection_frequencies(
+    rs::ResultSet,
+    iv_type::String;
+    n_loc_int::Int64=5,
+    ind_metrics::Vector{Int64}=_get_scen_ids(rs.ranks),
+)
+    selected_ranks = _get_iv_type(rs.ranks[scenarios=ind_metrics], iv_type)
+    ranks_frequencies = ranks_to_frequencies_ts(selected_ranks; n_ranks=n_loc_int)
+    loc_count = dropdims(sum(ranks_frequencies[ranks=1:n_loc_int], dims=[1, 3]), dims=3)[timesteps=1]
+
+    return loc_count
+end
+
+"""
+    summed_inverse_rank(ranks::NamedDimsArray, iv_type::String; dims=:scenarios,agg_func=x->x)
+    summed_inverse_rank(rs::ResultSet, iv_type::String; dims=:scenarios,agg_func=x->x)
+    summed_inverse_rank(ranks::NamedDimsArray; dims=:scenarios,agg_func=x->x)
+ 
+Calculates (number of sites) .- ranks summed over the dimension dims and transformed using agg_func 
+    (default no transformation).
```

Comment:
I've read this a few times now and I feel there's a simpler flow here.

You could have a `summed_inverse_rank` that does not have the `agg_func` argument, so it passes on `x -> x`.

I get the sense that a few responsibilities are being mixed together here. The flow might work better if:

1. You get the ranks of interest first
2. Pass it into this function for collation

```julia
iv_ranks = selection_ranks(rs, iv_type)  # this function is just renamed `_get_iv_type`

# AND/OR

iv_ranks = selection_ranks(ranks, iv_type)  # this function is just renamed `_get_iv_type`

ranks_of_interest = summed_inverse_rank(iv_ranks)
```

If you were to create a pipeline of processing steps:

```julia
ranks_of_interest = selection_ranks(rs, iv_type) |> 
    summed_inverse_ranks
```

This way you only have to support one use case.

I suppose the larger issue is I'm not sure how this is used or when it's useful - the function name and/or descriptor could be more informative - yes it's the summed inverse rank but for what purpose?
What alternate transformations/aggregations would be preferable?

And if you're supporting alternate aggregations, could the inversion not be one of them?

```julia
selection_ranks(rs, iv_type; agg_func=selection_inversion_function)
```

Reply:
This was a metric Pedro suggested because it meant the higher ranked the larger the number of summed inverse ranked. But you're right it could probably be an option in selection_ranks. Do you think it is useful? I think it could be a could measure but may be hard to explain to a wider audience.

Reply:
Oh I see, so it's not the ranks themselves, it's the rankings and the frequency they're at those ranks?

Reply:
Sort of but not entriely. For each rank it is (number of sites - rank), so the highest rank will have the highest values. Then it is summed over scenarios, so sites selected most frequently with the highest rank will have higher values than those selected less frequently at higher rank, if that makes sense.

Reply:
Alright - could you weight by the maximum rank to ease assessment? Values closer to 1.0 would then be locations that are most selected at higher ranks.